WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/usecases/InsightsAllTimeUseCase.kt

```diff
@@ -1,29 +1,41 @@
-package org.wordpress.android.ui.stats.refresh
+package org.wordpress.android.ui.stats.refresh.usecases
 
+import kotlinx.coroutines.experimental.CoroutineDispatcher
 import org.wordpress.android.R
 import org.wordpress.android.fluxc.model.SiteModel
 import org.wordpress.android.fluxc.model.stats.InsightsAllTimeModel
 import org.wordpress.android.fluxc.store.InsightsStore
+import org.wordpress.android.fluxc.store.StatsStore.InsightsTypes.ALL_TIME_STATS
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.ui.stats.refresh.BlockListItem
 import org.wordpress.android.ui.stats.refresh.BlockListItem.Empty
 import org.wordpress.android.ui.stats.refresh.BlockListItem.Item
 import org.wordpress.android.ui.stats.refresh.BlockListItem.Title
+import org.wordpress.android.ui.stats.refresh.Failed
+import org.wordpress.android.ui.stats.refresh.InsightsItem
+import org.wordpress.android.ui.stats.refresh.ListInsightItem
+import org.wordpress.android.ui.stats.refresh.toFormattedString
 import javax.inject.Inject
+import javax.inject.Named
 
 class InsightsAllTimeUseCase
-@Inject constructor(private val insightsStore: InsightsStore) {
-    suspend fun loadAllTimeInsights(site: SiteModel, refresh: Boolean, forced: Boolean): InsightsItem {
-        return if (refresh) {
-            val response = insightsStore.fetchAllTimeInsights(site, forced)
-            val model = response.model
-            val error = response.error
+@Inject constructor(
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
+    private val insightsStore: InsightsStore
+) : BaseInsightsUseCase(ALL_TIME_STATS, mainDispatcher) {
+    override suspend fun loadCachedData(site: SiteModel): InsightsItem? {
+        val dbModel = insightsStore.getAllTimeInsights(site)
+        return dbModel?.let { loadAllTimeInsightsItem(it) }
```

Comment:
`loadAllTimeInsightsItem()` handles a `null` model by displaying a "no data" message. Shouldn't `loadCachedData` always return a non-nullable `InsightsItem` this way?

Reply:
I think I'll try to make this consistent. If the model is null, I'll hide the block, if the data in the model is null, I'll show a block which has the "No data yet" message. The first use case should be logged as an error.